Source/WebCore/bindings/scripts/CodeGeneratorJS.pm

```diff
@@ -2121,6 +2121,65 @@ sub IsAlwaysExposedOnInterface
     return 1;
 }
 
+sub IsPrivateHeader
+{
+    my $headerFileName = shift;
+
+    # The set of DerivedSources headers that have private visibility.
+    # All private headers can only include other private headers or public headers.
+    # All private and public headers must be referenced using angle bracket includes.
+    # Always prefer using project visibility (instead of private) whenever possible.
+    my %privateHeaders = (
```

Comment:
This is unlikely to change frequently, but is there a way to collect this list by calling into xcodebuild somehow? (rather than maintaining it statically)

Reply:
Can we sort it at least?

Reply:
yes that I can do! (I started it sorted but then kinda lost lol will fix)

Reply:
> the script doesn't "know" about the Xcode build system

True, but also, this concept of private header teaches the script about the xcodebuild system... I wonder if we should move this to `CodeGeneratorJSExtensions+xcode.pm`, 

`use <config> qw(@IsPrivateHeader)`?

Reply:
I would like to see this list moved into another file. For those of us not using xcode, this seems like a landmine.

Also, can this be read from the module map (which will be cross-platform when we adopt swift maybe)?